plugins/woocommerce/src/Internal/ProductFilters/FilterData.php

```diff
@@ -0,0 +1,349 @@
+		$cached_data   = $this->get_cache( $transient_key );
+
+		if ( ! empty( $cached_data ) ) {
+			return $cached_data;
+		}
+
+		global $wpdb;
+		$stock_status_options = array_map( 'esc_sql', array_keys( wc_get_product_stock_status_options() ) );
+
+		add_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10, 2 );
+		add_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$query_vars['no_found_rows']  = true;
+		$query_vars['posts_per_page'] = -1;
+		$query_vars['fields']         = 'ids';
+		$query                        = new \WP_Query();
+
+		$query->query( $query_vars );
+		$product_query_sql = $query->request;
+
+		remove_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10 );
+		remove_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$stock_status_counts = array();
+
+		foreach ( $stock_status_options as $status ) {
+			$stock_status_count_sql = "
+				SELECT COUNT( DISTINCT posts.ID ) as status_count
+				FROM {$wpdb->posts} as posts
+				INNER JOIN {$wpdb->postmeta} as postmeta ON posts.ID = postmeta.post_id
+				AND postmeta.meta_key = '_stock_status'
+				AND postmeta.meta_value = '{$status}'
+				WHERE posts.ID IN ( {$product_query_sql} )
+			";
+
+			$result = $wpdb->get_row( $stock_status_count_sql ); // phpcs:ignore
+			$stock_status_counts[ $status ] = $result->status_count;
+		}
+
+		$this->set_cache( $transient_key, $stock_status_counts );
+
+		return $stock_status_counts;
+	}
+
+	/**
+	 * Get rating counts for the current products.
+	 *
+	 * @param array $query_vars The WP_Query arguments.
+	 * @return array rating=>count pairs.
+	 */
+	public function get_rating_counts( array $query_vars ) {
+		$pre_filter_counts = $this->pre_get_filter_counts( 'rating', $query_vars );
+
+		if ( isset( $pre_filter_counts ) ) {
+			return $pre_filter_counts;
+		}
+
+		$transient_key = $this->get_transient_key( $query_vars, 'rating' );
+		$cached_data   = $this->get_cache( $transient_key );
+
+		if ( ! empty( $cached_data ) ) {
+			return $cached_data;
+		}
+
+		global $wpdb;
+
+		add_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10, 2 );
+		add_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$query_vars['no_found_rows']  = true;
+		$query_vars['posts_per_page'] = -1;
+		$query_vars['fields']         = 'ids';
+		$query                        = new \WP_Query();
+
+		$query->query( $query_vars );
+		$product_query_sql = $query->request;
+
+		remove_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10 );
+		remove_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$rating_count_sql = "
+			SELECT COUNT( DISTINCT product_id ) as product_count, ROUND( average_rating, 0 ) as rounded_average_rating
+			FROM {$wpdb->wc_product_meta_lookup}
+			WHERE product_id IN ( {$product_query_sql} )
+			AND average_rating > 0
+			GROUP BY rounded_average_rating
+			ORDER BY rounded_average_rating DESC
+		";
```

Comment:
I'm wondering if here it would be the case to also use `$wpdb->prepare`:

```suggestion		
                $rating_count_sql = $wpdb->prepare(
			"
			SELECT COUNT( DISTINCT product_id ) as product_count, ROUND( average_rating, 0 ) as rounded_average_rating
			FROM {$wpdb->wc_product_meta_lookup}
			WHERE product_id IN ( %s )
			AND average_rating > 0
			GROUP BY rounded_average_rating
			ORDER BY rounded_average_rating DESC
			",
			$product_query_sql
		);
```